Classes/Emogrifier.php

```diff
@@ -1209,19 +1209,17 @@ private function parseCssDeclarationsBlock($cssDeclarationsBlock)
         }
 
         $properties = [];
+        $declarations = preg_split('/;(?!base64|charset)/', $cssDeclarationsBlock);
 
-        // Replace the semicolon in base64-encoded data URIs with a @ to prevent breaking
-        // up these declarations.
-        $mangledDeclarationsBlock = str_replace(';base64', '@base64', $cssDeclarationsBlock);
-        $declarations = explode(';', $mangledDeclarationsBlock);
         foreach ($declarations as $declaration) {
+            $declaration = trim($declaration);
```

Comment:
Please don't overwrite variables because then the same variable can have two different meanings depending on how far down in the method it is used. This makes understanding the method harder and also makes moving code around harder.

Proposed solution 1: use a new variable $trimmedDeclaration
Proposed solution 2: inline $declaration in the preg_match (the only place where it is used AFAICS)
Proposed solution 3: drop the trim altogether and modify the preg_match to also cope with leading and trailing whitespace